praw/objects.py

```diff
@@ -765,22 +765,30 @@ class Redditor(Gildable, Messageable, Refreshable):
     def __init__(self, reddit_session, user_name=None, json_dict=None,
                  fetch=False):
         """Construct an instance of the Redditor object."""
+        if not user_name:
+            user_name = json_dict['name']
```

Comment:
Was this broken in some cases before if user_name wasn't provided? If it was, we should probably add a few tests pertaining to Redditor info url.